cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ConfigMapOperator.java

```diff
@@ -29,4 +32,22 @@ public ConfigMapOperator(Vertx vertx, KubernetesClient client) {
     protected MixedOperation<ConfigMap, ConfigMapList, DoneableConfigMap, Resource<ConfigMap, DoneableConfigMap>> operation() {
         return client.configMaps();
     }
+
+    @Override
+    protected Future<ReconcileResult<ConfigMap>> internalPatch(String namespace, String name, ConfigMap current, ConfigMap desired) {
+        try {
+            TreeMap a = new TreeMap(current.getData());
+            TreeMap b = new TreeMap(desired.getData());
+            if (a.toString().equals(b.toString())) {
```

Comment:
You can just use [`.equals()` on the maps](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#equals-java.lang.Object-), no need to convert them to Strings. This also means there's no need to copy the maps into `TreeMaps` for consistent ordering.